src/ResourceManager/Sql/Commands.Sql/Database/Services/AzureSqlDatabaseAdapter.cs

```diff
@@ -286,8 +286,56 @@ internal IEnumerable<AzureSqlDatabaseActivityModel> ListDatabaseActivity(string
             }
             else
             {
-                throw new NotSupportedException(string.Format(CultureInfo.InvariantCulture, Microsoft.Azure.Commands.Sql.Properties.Resources.StandaloneDatabaseActivityNotSupported));
+                var response = Communicator.ListOperations(resourceGroupName, serverName, databaseName);
+                IEnumerable<AzureSqlDatabaseActivityModel> list = response.Select((r) =>
+                {
+                    return new AzureSqlDatabaseActivityModel()
+                    {
+                        DatabaseName = r.DatabaseName,
+                        ErrorCode = r.ErrorCode,
+                        ErrorMessage = r.ErrorDescription,
+                        ErrorSeverity = r.ErrorSeverity,
+                        Operation = r.Operation,
+                        OperationId = Guid.Parse(r.Name),
+                        PercentComplete = r.PercentComplete,
+                        ServerName = r.ServerName,
+                        StartTime = r.StartTime,
+                        State = r.State,
+                        Properties = new AzureSqlDatabaseActivityModel.DatabaseState()
+                        {
+                            Current = new Dictionary<string, string>(),
+                            Requested = new Dictionary<string, string>()
+                        }
+                    };
+                });
+
+                // Check if we have a operation id constraint
+                if (operationId.HasValue)
+                {
+                    list = list.Where(pl => Guid.Equals(pl.OperationId, operationId));
```

Comment:
it would be better to do this filter before doing the select.

Reply:
not fixed at this moment and keep it same pattern as above due to not many "updateslo" operations